airflow/providers/airbyte/CHANGELOG.rst

```diff
@@ -24,6 +24,25 @@
 Changelog
 ---------
 
+3.2.0
+.....
+
+This release of provider is only available for Airflow 2.3+ as explained in the Apache Airflow
+providers support policy https://github.com/apache/airflow/blob/main/README.md#support-for-providers
+
+Misc
+~~~~
+
+* ``Move min airflow version to 2.3.0 for all providers (#27196)``
```

Comment:
Yes I think we should because otherwise we will have to selectively do it every time when we will track any change that is 2.3 incompatible. It makes it much cleaner and easier to manage and creates far less overhead when preparing the release wave. And It gives the user no more burden - most of the users are upgrading together with Airlfow - which means that they will automatically use the new providers.

It also provides a much better starting point in the future when we will have to (for example for security reasons) release a catastrofic fix in a proivider that should be installed for pre-2.2 version. We know exactly where to branch off, we still have patch-levels to use for that.

Most of our users do not follow and release providers when they are released (and we don't even encourage it). The big benefit of releasing provideres independently, is that the users CAN do that when they want. 

Also releasing it now gives a strong signal - there will be no more feature releases for all providers with pre-2.3 version. There will be at most bugfixes. If we don't upgrade now, there is still an option to release feature version for those providers that is not 2.2 compatible, which we shoudl not do any more. And this is yet another signal to our users that they should upgrade to Airflow 2.3 asap. 

So yes - I think releasing all providers now makes perfect sense. Mostly that this is a clear communication "no more feature provider releases that will support Airflow 2.2"

Reply:
But - just to comment on that as well @jedcunningham  - I thought also why I am doing it this way.

There is another good reason why we should release them all at least this time (but we can think about changing it in the future). 

Releasing "all" providers in such cases is simply much simpler and less risky and error-prone for me as release manager. Currently you would have to manually go through pretty much all those packages and find out where minor upgrade is beacuase we have "bump" and where there are "real features".

It's just way more work for me and the kind of work is far more prone to manual mistakes (unless we automate this scenario somehow, which makes it far more stresful - because being perfectiionist, I will likely have to check it 4 or five times if I made any manual mistake or not.  And knowing myself some mistake will slip through anyway. I am simply human.

So If we are to do that, I would likely have to first implement some automation to do it, otherwise I will find myself in a situation that I do not really like, where I have a working, automated process which has a number of safeguards in place that will prevent from making stupid mistakes. And likely it's better to do it for the next wave.

Reply:
Just to give a very good example of propblems if we do not release "all" providers. 
Real example from just now. Before rebase I am doing now) someone merged:

* ``Replace urlparse with urlsplit (#27389)``

This change is a non-breaking, non feature, "misc" kind of change, that we - regardless - shoud put in the docs, just in case this caused some problems or changes we are not aware of (it's not a bugfix, not a feature, but it changes code so potentially we should be able to explain that it was the reason when someone has some problems.

But this change was touching 7 providers. This means that it *could* have changed whether we release a provider or no (previously it was min-version only, now it became min-version + urlparse).

Should we release it now ? I think yes - the longer we wait with realeasing any code change the better. If we hold with relasing of provider that already has some "code" change" - then if we release it in 3 months we might have detect some problems that we could hve detected 3 months earlier. 

By having "release all changed in any way except documentation" providers I free myself from having to review each of those manually and decide - I release this/I do not release that. Currently we automatically skip from release only providers that have "doc-only changes" - for those we only regenerate the documentation. And this is all automated. If we were to decide "and we do not realease also that change" - we would have to implement automation for exclusion of those changes.

Reply:
Simmply the more manual decision points, the less automation - the more "mistakes" we will do when releasing NN+ providers at the same time.

Reply:
Thanks @potiuk, yep that all makes sense. I was simply looking at if from the users side initially, but I definitely see value in keeping it simple for us.

Reply:
> Thanks @potiuk, yep that all makes sense. I was simply looking at if from the users side initially, but I definitely see value in keeping it simple for us.

I knew you would :)